### ISSUE TITLE
Error "unmatched close parenthesis" (compile_key_pattern) when running `i18n-tasks health`

### ISSUE THREAD

**Issue by lucascaton** [glebm/i18n-tasks#180]
When I run `$ i18n-tasks health`, I got this result:

```
`Forest (en, pt-BR) has 274 keys across 2 locales. On average, values are 15 characters long, keys have 3.1 segments, a locale has 137 keys.
✓ Well done! No translations are missing.

~/.rvm/gems/ruby-2.2.3/gems/i18n-tasks-0.9.2/lib/i18n/tasks/key_pattern_matching.rb:22:in `compile_key_pattern': unmatched close parenthesis: /\A((?<=^|\.)[^.]+?(?=\.|$)|'\.'))\.(?<=^|\.)[^.]+?(?=\.|$)\.title,landing_pages\.index\.features_(?<=^|\.)[^.]+?(?=\.|$),landing_pages\.index\.features_(?<=^|\.)[^.]+?(?=\.|$)_description,landing_pages\.index\.platform_(?<=^|\.)[^.]+?(?=\.|$),landing_pages\.index\.platform_(?<=^|\.)[^.]+?(?=\.|$)_description,landing_pages\.index\.footer-(?<=^|\.)[^.]+?(?=\.|$)_version}\z/ (RegexpError)
```

* Mac OS X 10.11.1
* Ruby 2.2.3
* Rails 4.2.5
* i18n-tasks (gem) 0.9.2

**Comment by glebm** [comment]
Can you post your config/i18n-tasks.yml?

**Comment by lucascaton** [comment]
@glebm Sure!

**config/i18n-tasks.yml**
```yaml
base_locale: en

data:
  read:
    - config/locales/**/%{locale}.yml
    - config/locales/**/*.%{locale}.yml

search:
  exclude:
    - app/assets/images
    - app/assets/fonts
```

My locale files:

```
config/locales
├── activerecord.en.yml
├── activerecord.pt-BR.yml
├── enumerations.en.yml
├── enumerations.pt-BR.yml
└── views
    ├── calendars.en.yml
    ├── calendars.pt-BR.yml
    ├── categories.en.yml
    ├── categories.pt-BR.yml
    ├── dashboard.en.yml
    ├── dashboard.pt-BR.yml
    ├── devise.en.yml
    ├── devise.pt-BR.yml
    ├── en.yml
    ├── landing_pages.en.yml
    ├── landing_pages.pt-BR.yml
    ├── pt-BR.yml
    ├── settings.en.yml
    ├── settings.pt-BR.yml
    ├── transactions.en.yml
    └── transactions.pt-BR.yml
```

**Comment by lucascaton** [comment]
@glebm hi there. Any clue how to fix it? Thanks :smile:

**Comment by glebm** [comment]
Looks like there is a dynamic key usage in the code with a key that looks something like below that i18n-tasks has trouble with:

```ruby
t('landing_pages.index.footer-#{stuff}_version')
```

Can you post the exact code of that usage here? Thanks

**Comment by lucascaton** [comment]
Yep, this the part of my haml where I'm using it:

```
- Locale.list.each do |locale|
  = link_to t(".footer-#{locale}_version", locale: locale), locale: locale
```

**Comment by glebm** [comment]
Looks like that's not the key that's causing the issue. Could you perhaps inspect the output of `i18n-tasks find --no-strict` and see if there is anything fishy in there?

**Comment by lucascaton** [comment]
Sure!

`i18n-tasks find --no-strict`: https://gist.github.com/lucascaton/ccd664189febc3005bd6
`i18n-tasks find --no-strict -fkeys`: https://gist.github.com/lucascaton/ccd664189febc3005bd6

**Comment by glebm** [comment]
The commit above should fix it

**Comment by lucascaton** [comment]
It worked, thanks a lot! :smile:

**Comment by lucascaton** [comment]
@glebm could you please bump a new version? Thanks again!

**Comment by glebm** [comment]
@lucascaton Released in v0.9.3.

**Comment by lucascaton** [comment]
@glebm thanks!
